MoreLinq.Test/SubsetTest.cs

```diff
@@ -24,25 +24,29 @@ public void TestSubsetsIsLazy()
         /// Verify that negative subset sizes result in an exception.
         /// </summary>
         [Test]
-        [ExpectedException(typeof(ArgumentOutOfRangeException))]
         public void TestNegativeSubsetSize()
         {
-            const int count = 10;
-            var sequence = Enumerable.Range(1, count);
-            sequence.Subsets(-5);
+            Assert.Throws<ArgumentOutOfRangeException>(() =>
+            {
+                const int count = 10;
+                var sequence = Enumerable.Range(1, count);
+                sequence.Subsets(-5);
+            });
         }
 
         /// <summary>
         /// Verify that requesting subsets larger than the original sequence length result in an exception.
         /// </summary>
         [Test]
-        [ExpectedException(typeof(ArgumentOutOfRangeException))]
         public void TestSubsetLargerThanSequence()
         {
-            const int count = 10;
-            var sequence = Enumerable.Range(1, count);
-            var result = sequence.Subsets(count + 5);
-            result.Count(); // this particular exception is deferred until sequence evaluation
+            Assert.Throws<ArgumentOutOfRangeException>(() =>
+            {
+                const int count = 10;
+                var sequence = Enumerable.Range(1, count);
+                var result = sequence.Subsets(count + 5);
```

Comment:
Only this line should be inside the lambda to Assert.Throws as that's what will cause and being tested for an exception.